### ISSUE TITLE
Remove legacy h5py based formats

### ISSUE THREAD

**Issue by benjeffery** [tskit-dev/tskit#2812]
Opening this up for discussion.

As mentioned by @jeromekelleher at https://github.com/tskit-dev/tskit/pull/2811#issuecomment-1663778875 it has been a long time since these legacy formats were used. When dropping them we should a add a note in the docs to specify the version of tskit to use to upgrade the file.

**Comment by jeromekelleher** [comment]
Agree. We should add a stable hyperlink on the tskit docs (Legacy HDF5 file formats) which we provide a link to in the error message also, so people know where to look when the file doesn't load.

**Comment by molpopgen** [comment]
I think we should do this ASAP.  Definitely want to get it out of the way before any kind of 1.0 release is on the horizon.

**Comment by jeromekelleher** [comment]
I might just go ahead and do this, we've got a bunch of PRs in flight and #2811 doesn't look like a quick fix. Any objections @benjeffery?

**Comment by benjeffery** [comment]
#2811 needs to be done for `lshmm` anyway. So expediting this won't help. I should get #2811done today.

**Comment by jeromekelleher** [comment]
Ah right, let's backburner so
